src/store/ui/types.ts

```diff
@@ -0,0 +1,32 @@
+/*
+*  File:            ui/types.ts
+*  Description:     Holds types and constants for managing global UI states
+*  Author:          ChRIS ui Demo                 
+*/
+
+
+import keyMirror from "keymirror";
+
+// Description: state for UI Manager
+export interface IUiState {
```

Comment:
For UI component state, can we keep the state in the component, and leave the app wide states here? Is there a benefit of tracking ui state in redux?

Reply:
My preference is to keep state that is used in more than one component centralized in the redux store. There is really no "right" answer for this specific question, just best practice and dev preferences (Ref: https://redux.js.org/faq/organizing-state#do-i-have-to-put-all-my-state-into-redux-should-i-ever-use-reacts-setstate). Generally, the only time I use local state is in things like form components. Otherwise, my preference is to have it centralized in the store.

Reply:
I think my preference would be otherwise, but lets go with what you have for now.